src/ipc-wrapper/client.js

```diff
@@ -0,0 +1,72 @@
+// @ts-check
+import { createClient } from 'rpc-reflector'
+import QuickLRU from 'quick-lru'
+import { MANAGER_CHANNEL_ID, SubChannel } from './sub-channel.js'
+
+const PROJECT_CLIENT_MAX_AGE = 1000 * 60 * 60 // 1 hour
+
+/**
+ * @param {import('./sub-channel.js').MessagePortLike} messagePort
+ * @returns {import('rpc-reflector/client.js').ClientApi<import('../mapeo-manager.js').MapeoManager>}
+ */
+export function createMapeoClient(messagePort) {
+  /** @type {QuickLRU<import('../types.js').ProjectPublicId, { instance: import('rpc-reflector/client.js').ClientApi<import('../mapeo-project.js').MapeoProject>, channel: SubChannel }>} */
+  const existingProjectClients = new QuickLRU({
+    maxSize: 3,
+    maxAge: PROJECT_CLIENT_MAX_AGE,
+    onEviction: (_id, { instance, channel }) => {
+      createClient.close(instance)
+      channel.close()
+    },
+  })
+
+  const managerChannel = new SubChannel(messagePort, MANAGER_CHANNEL_ID)
+
+  /** @type {import('rpc-reflector').ClientApi<import('../mapeo-manager.js').MapeoManager>} */
+  const managerClient = createClient(managerChannel)
+
+  managerChannel.start()
+
+  const client = new Proxy(managerClient, {
+    get(target, prop, receiver) {
+      if (prop === 'getProject') {
+        return createProjectClient
+      } else {
+        return Reflect.get(target, prop, receiver)
+      }
+
+      /**
+       * @param {import('../types.js').ProjectPublicId} projectPublicId
+       * @returns {Promise<import('rpc-reflector/client.js').ClientApi<import('../mapeo-project.js').MapeoProject>>}
+       */
+      function createProjectClient(projectPublicId) {
```

Comment:
Move this out of the scope of `get()`, into the `createMapeoClient()` scope.

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/261/commits/fd58298a81bb2dae703bcd11ec76213b41d19863